lib/src/main/java/graphql/nadel/engine/blueprint/NadelSchemaTraverserElement.kt

```diff
@@ -0,0 +1,247 @@
+package graphql.nadel.engine.blueprint
+
+import graphql.nadel.engine.util.unwrapAll
+import graphql.schema.GraphQLAppliedDirective
+import graphql.schema.GraphQLAppliedDirectiveArgument
+import graphql.schema.GraphQLArgument
+import graphql.schema.GraphQLDirective
+import graphql.schema.GraphQLEnumType
+import graphql.schema.GraphQLEnumValueDefinition
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLFieldsContainer
+import graphql.schema.GraphQLInputFieldsContainer
+import graphql.schema.GraphQLInputObjectField
+import graphql.schema.GraphQLInputObjectType
+import graphql.schema.GraphQLInputType
+import graphql.schema.GraphQLInterfaceType
+import graphql.schema.GraphQLNamedSchemaElement
+import graphql.schema.GraphQLNamedType
+import graphql.schema.GraphQLObjectType
+import graphql.schema.GraphQLOutputType
+import graphql.schema.GraphQLScalarType
+import graphql.schema.GraphQLUnionType
+
+internal sealed interface NadelSchemaTraverserElement {
```

Comment:
Model for traversals, instead of having

```kotlin
        fun visitGraphQLScalarType(
            parent: GraphQLNamedSchemaElement?,
            node: GraphQLScalarType,
        ): Boolean
```

we have

```kotlin
        fun visitGraphQLScalarType(
            element: NadelSchemaTraverserElement.Scalar
        ): Boolean
```

Where it is known that `Scalar` type doesn't have a parent.

Another e.g.

```kotlin
    data class FieldDefinition(
        val parent: GraphQLFieldsContainer,
        override val node: GraphQLFieldDefinition,
    ) : NadelSchemaTraverserElement
```

It's better modelled here that a field has a parent field container.